lib/electric_slide/call_queue.rb

```diff
@@ -288,20 +288,20 @@ def remove_call(call)
     # @param [Agent] agent Agent to be connected
     # @param [Adhearsion::Call] call Caller to be connected
     def connect(agent, queued_call)
-      unless queued_call && queued_call.active?
+      if queued_call && queued_call.active?
+        queued_call[:agent] = agent
+
+        logger.info "Connecting #{agent} with #{remote_party queued_call}"
+        case @connection_type
+        when :call
+          call_agent agent, queued_call
+        when :bridge
+          bridge_agent agent, queued_call
+        end
+      else
         logger.warn "Inactive queued call found in #connect"
         return_agent agent
       end
-
```

Comment:
Moving all this into the conditional isn't necessary. The method should bail out after `return_agent`.  To keep things concise, I'd rather see this:

```Ruby
unless queued_call && queued_call.active?
  logger.warn "Inactive queued call found in #connect"
  return_agent agent
  return
end

# ... rest of method
```